### PR TITLE
Elevator reset button & elevator eye candy

### PR DESCRIPTION
**Pull request by marredal**
Added reset button for Lights Out which puts all the buttons back to their starting position.
Added meshes to make elevator scene look good.
Moved cylinder shape for bucket puzzle.

### CODE REVIEW

File: Game/Resources/elevatorScene/Scripts/LightsOutScript.as [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class LightsOutScript {
     Entity @board;
     Entity @rightController;
-    Entity @elevatorCartScript;
+	Entity @elevatorCartScript;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/LightsOutScript.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/LightsOutScript.as [hunk 2]
```diff
@@ -57,6 +57,18 @@ class LightsOutScript {
...
+	void ResetBoard() {
+		Entity @btn;
+        if(Input(Trigger, rightController)) {
+		 for (int x = 0; x < 16; ++x) {
+			buttonStates["" + x] = false;
+			buttons.get("" + x, @btn);
+			btn.position.x = 0.0f;
+            numPressedButtons = 0;
+            }
+		 }
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/LightsOutScript.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/LightsOutScript.as [hunk 3]
```diff
@@ -57,6 +57,18 @@ class LightsOutScript {
         return index >= 0 && index <= 15;
     }
 
+	void ResetBoard() {
+		Entity @btn;
+        if(Input(Trigger, rightController)) {
+		 for (int x = 0; x < 16; ++x) {
+			buttonStates["" + x] = false;
+			buttons.get("" + x, @btn);
+			btn.position.x = 0.0f;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/LightsOutScript.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/LightsOutScript.as [hunk 4]
```diff
@@ -57,6 +57,18 @@ class LightsOutScript {
         return index >= 0 && index <= 15;
     }
 
+	void ResetBoard() {
+		Entity @btn;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/LightsOutScript.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as [hunk 5]
```diff
@@ -1,57 +1,57 @@
...
-    bool openDoor;
-    vec3 tempPosWithX;
-    vec3 tempPosWithService;
-    float speed;
-    
+	Entity @doorWithService;
+	bool openDoor;
+	vec3 tempPosWithX;
+	vec3 tempPosWithService;
+	float speed;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as [hunk 6]
```diff
@@ -1,57 +1,57 @@
...
 
     elevator_backDoorCon(Entity @entity){
         @hub = Managers();
         @doorWithX = GetEntityByGUID(1511870598);
-        @doorWithService = GetEntityByGUID(1511870668);
-        openDoor = false;
-        speed = 0.5f;
+		@doorWithService = GetEntityByGUID(1511870668);
+		openDoor = false;
+		speed = 0.5f;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as [hunk 7]
```diff
@@ -1,57 +1,57 @@
...
+				tempPosWithX = doorWithX.GetWorldPosition();
+				tempPosWithService = doorWithService.GetWorldPosition();
+			
+				tempPosWithX.z -= speed * deltaTime;
+				tempPosWithService.z += speed * deltaTime;
+				
+				doorWithX.SetWorldPosition(tempPosWithX);
+				doorWithService.SetWorldPosition(tempPosWithService);
+			}
+		}
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as]: Tabs,

File: Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as [hunk 8]
```diff
@@ -1,23 +1,23 @@
...
-    vec3 tempPosWithService;
-    float boardPitch;
-    float speed;
+	Entity @doorWithService;
+	Entity @puzzleBoard;
+	bool openDoor;
+	vec3 tempPosWithX;
+	vec3 tempPosWithService;
+	float boardPitch;
+	float speed;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as [hunk 9]
```diff
@@ -26,46 +26,46 @@ class elevator_frontDoorCon {
...
+				tempPosWithX = doorWithX.GetWorldPosition();
+				tempPosWithService = doorWithService.GetWorldPosition();
+			
+				tempPosWithX.z += speed * deltaTime;
+				tempPosWithService.z -= speed * deltaTime;
+				
+				doorWithX.SetWorldPosition(tempPosWithX);
+				doorWithService.SetWorldPosition(tempPosWithService);
+			}
+		}
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_moveCart.as [hunk 10]
```diff
@@ -69,13 +71,13 @@ class elevator_moveCart {
...
-        SendMessage(board, 0);
-    }
+	
+	void MoveForward() {
+		phase = 0;
+	}
+	
+	void StopCart() {
+        phase = 4;
+		SendMessage(board, 0);
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_moveCart.as [hunk 11]
```diff
@@ -1,30 +1,31 @@
...
         @puzzleBoard = GetEntityByGUID(1512029307);
+
         
         @realSelf = @entity;
         realSelf.SetEnabled(false, true);
         
-        moveForward = true;
-        moveUpward = false;
+		moveForward = true;
+		moveUpward = false;
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_moveCart.as [hunk 12]
```diff
@@ -1,30 +1,31 @@
...
     int phase;
     float uniformScale;
 
     elevator_moveCart(Entity @entity){
         @hub = Managers();
         @self = GetEntityByGUID(1508919163);
-        @elevator = GetEntityByGUID(1511870044);
-        @board = GetEntityByGUID(1511530025);
+		@elevator = GetEntityByGUID(1511870044);
+		@board = GetEntityByGUID(1511530025);
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: Tabs.

File: Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as [hunk 13]
```diff
@@ -26,46 +26,46 @@ class elevator_frontDoorCon {
...
     }
+	
+	void OpenDoor() {
+		openDoor = true;
+	}
+	
+	void CloseDoor() {
+		openDoor = false;
+	}
```
Comment (ludvigarlebrink) [inline Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as]: Tabs.

**Comment by andens** [comment]
iamheman

**Comment by marredal** [comment]
what does this mean?

**Comment by andens** [comment]
@marredal It makes the build servers rebuild the project :) In this case an internal debug thing on Jenkins that prevented the build from succeeding was removed, so I triggered a rebuild to see if it worked out.

**Pull request merged**